### PR TITLE
[3446] added isSortSelect property and modified styling to show chevron icon on mobile

### LINKED ISSUE #3446: Arrow near 'Sort' on PLP is missing on mobile
**Description**:
Arrow near 'Sort' on PLP is missing on mobile

**Expected behavior**:
Arrow near 'Sort' on PLP is present on mobile

**Screenshots, Video, Logs**
image

**To Reproduce**:
1. Open any PLP on mobile

**Actual result**:
Arrow near 'Sort' on PLP is missing on mobile

### PR DESCRIPTION
**Pull request by pablobentivengo**
**Related issue(s):**
* Fixes #3446 

**Problem:**
* Arrow near 'Sort' on PLP is missing on mobile

**In this PR:**
* Added isSortSelect prop to Field and FieldSelect
* FieldSelect component conditionally renders "sort" if isSortSelect
* Modified CategorySelect styling to show chevron (arrow) on mobile

### CODE REVIEW

**Comment by aleksandrakorolova** [comment]
1. Arrow is shifter up on iOS but it's shifted slightly down on Android. Please, align the arrow ± in the center on both platforms
2. Inverted arrow is displayed when selecting sorting method
https://drive.google.com/file/d/10upE16Ncb3gdxoL3f8cJ0TlqwJ6n1ZzH/view?usp=sharing

**Comment by pablobentivengo** [comment]
Not ready

**Comment by pablobentivengo** [comment]
2) With @Elena-Evsa and @carinadues we agreed to create a new issue for the arrow not changing direction properly on mobile devices, since this issue is present in all dropdowns. The issue is generated because of the way the dropdown is rendered on mobile devices. Arrow only changes direction when opening dropdown.

_Tests done on PMREV_
- IOS
 https://vimeo.com/680814098/146639adee
- Android
 https://vimeo.com/680814806/99e34d1cc2

**Comment by aleksandrakorolova** [comment]
1. Fixed
2. Ok
3. Please, change text 'Sort' to 'Sort by' and change font size to 16px as 'Filter'. Also, add margin 10px between text 'Sort by' and arrow as between Filter icon and 'Filter text'

File: packages/scandipwa/src/component/CategorySort/CategorySort.style.scss [hunk 1]
```diff
@@ -59,6 +59,39 @@
...
+            .ChevronIcon {
+                @include mobile {
+                    position: relative;
+                    inset-inline-end: 0;
+                    margin-inline-start: 10px;
+
+                    @supports (-webkit-touch-callout: none) {
+                        /* CSS specific to iOS devices */ 
+                        position: absolute;
+                        margin-inline-start: 0px;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CategorySort/CategorySort.style.scss:87]: Stay constantly. 0 goes without px

File: packages/scandipwa/src/component/CategorySort/CategorySort.style.scss [hunk 2]
```diff
@@ -82,21 +114,17 @@
...
-            }
+    .Field { 
+        &-LabelContainer {
+            .ChevronIcon {
+                @include desktop {
+                    display: none;
+                }
 
-            margin-block-start: 3px;
+                margin-block-start: 3px;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CategorySort/CategorySort.style.scss:126]: first properties, then new selectors

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.component.js [hunk 3]
```diff
@@ -36,7 +36,12 @@ export class FieldSelect extends PureComponent {
...
         isDisabled: PropTypes.bool.isRequired,
         isUpDirection: PropTypes.bool.isRequired,
-        isScrollable: PropTypes.bool.isRequired
+        isScrollable: PropTypes.bool.isRequired,
+        isSortSelect: PropTypes.bool
+    };
+
+    static defaultProps = {
+        isSortSelect: false
     };
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/FieldSelect/FieldSelect.component.js:45]: So isn't always passed from field component? It is most lucky always will be with default value.

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.component.js [hunk 4]
```diff
@@ -159,6 +164,18 @@ export class FieldSelect extends PureComponent {
...
+    renderSortSelect() {
+        const { isSortSelect } = this.props;
+
+        if (isSortSelect) {
+            return (
+                <div block="FieldSelect" elem="SortSelect">{ __('Sort by') }</div>
+            );
+        }
+
+        return null;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/FieldSelect/FieldSelect.component.js:176]: Replace them

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/CategorySort/CategorySort.style.scss (hunk 2 above)
Comment (pablobentivengo) [reply packages/scandipwa/src/component/CategorySort/CategorySort.style.scss]: done

File: packages/scandipwa/src/component/CategorySort/CategorySort.style.scss (hunk 1 above)
Comment (pablobentivengo) [reply packages/scandipwa/src/component/CategorySort/CategorySort.style.scss]: done

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js [hunk 5]
```diff
@@ -35,12 +35,14 @@ export class FieldSelectContainer extends PureComponent {
...
-        changeValueOnDoubleClick: PropTypes.bool
+        changeValueOnDoubleClick: PropTypes.bool,
+        isSortSelect: PropTypes.bool
     };
 
     static defaultProps = {
         noPlaceholder: false,
-        changeValueOnDoubleClick: false
+        changeValueOnDoubleClick: false,
+        isSortSelect: false
```
[the remark this answers is not in the document]
Comment (pablobentivengo) [reply packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js:45]: done

**Review by AleksandrsKondratjevs — APPROVED** [review]

**Pull request merged**